From fd74f7def27bb8763fe5e4fdc569698d2c75eaa2 Mon Sep 17 00:00:00 2001
From: Benjamin Hipple <bhipple@protonmail.com>
Date: Sat, 25 Apr 2020 22:20:16 -0400
Subject: [PATCH] Do not check version

---
 qt/aqt/main.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/qt/aqt/main.py b/qt/aqt/main.py
index 3f3034bb..ff944d4d 100644
--- a/qt/aqt/main.py
+++ b/qt/aqt/main.py
@@ -1122,6 +1122,9 @@ title="%s" %s>%s</button>""" % (
     ##########################################################################
 
     def setupAutoUpdate(self) -> None:
+        # Don't check for latest version since the versions are
+        # managed in Nixpkgs.
+        return
         import aqt.update
 
         self.autoUpdate = aqt.update.LatestVersionFinder(self)
-- 
2.15.2

